hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -120,7 +120,7 @@ private AbfsClient(final URL baseUrl, final SharedKeyCredentials sharedKeyCreden
     this.abfsCounters = abfsClientContext.getAbfsCounters();
 
     this.executorService = MoreExecutors.listeningDecorator(
-        Executors.newScheduledThreadPool(this.abfsConfiguration.getNumLeaseThreads()));
+        HadoopExecutors.newScheduledThreadPool(this.abfsConfiguration.getNumLeaseThreads()));
```

Comment:
can you make sure this executor marks its threads as daemons. Otherwise processes can hang during shutdown. @bgaborg has encountered this elsewhere